### PR TITLE
Prevent invalid voting

### PR DESCRIPTION
**Pull request by tallywiesenberg**
Previously, an account could call `vote()` on a dispute that had closed. I added a require block on the vote function to revert the `vote()` call if the account is voting on a dispute that has closed.

### CODE REVIEW

File: contracts/Governance.sol [hunk 1]
```diff
@@ -88,12 +88,13 @@ contract Governance is TellorVars{
...
         require(IOracle(_oracle).getBlockNumberByTimestamp(_requestId, _timestamp) != 0, "Mined block is 0");
         address _reporter = IOracle(_oracle).getReporterByTimestamp(_requestId,_timestamp);
-        bytes32 _hash = keccak256(abi.encodePacked(_requestId, _timestamp));
+        bytes32 _hash = keccak256(abi.encodePacked(_requestId, _timestamp, _reporter));
         voteCount++;
         uint256 _id = voteCount;
         voteRounds[_hash].push(_id);
         if (voteRounds[_hash].length > 1) {
             uint256 _prevId = voteRounds[_hash][voteRounds[_hash].length - 2];
+            require(block.timestamp - voteInfo[_prevId].startDate > 12 hours, "this dispute is still open");
```
Comment (themandalore) [inline contracts/Governance.sol:97]: I don't think this is needed.  You can't tally the vote if it's under the vote time right? So the next require statement should grab it

File: contracts/Governance.sol [hunk 2]
```diff
@@ -88,12 +88,13 @@ contract Governance is TellorVars{
         address _oracle = IController(TELLOR_ADDRESS).addresses(_ORACLE_CONTRACT);
         require(IOracle(_oracle).getBlockNumberByTimestamp(_requestId, _timestamp) != 0, "Mined block is 0");
         address _reporter = IOracle(_oracle).getReporterByTimestamp(_requestId,_timestamp);
-        bytes32 _hash = keccak256(abi.encodePacked(_requestId, _timestamp));
+        bytes32 _hash = keccak256(abi.encodePacked(_requestId, _timestamp, _reporter));
```
Comment (themandalore) [inline contracts/Governance.sol:91]: You don't want to hash the reporter, the dispute shouldn't matter who it's started from

File: contracts/Governance.sol (hunk 1 above)
Comment (tallywiesenberg) [reply contracts/Governance.sol:97]: Agreed thanks for pointing this out

File: contracts/Governance.sol [hunk 3]
```diff
@@ -88,13 +88,12 @@ contract Governance is TellorVars{
...
         require(IOracle(_oracle).getBlockNumberByTimestamp(_requestId, _timestamp) != 0, "Mined block is 0");
         address _reporter = IOracle(_oracle).getReporterByTimestamp(_requestId,_timestamp);
-        bytes32 _hash = keccak256(abi.encodePacked(_requestId, _timestamp, _reporter));
+        bytes32 _hash = keccak256(abi.encodePacked(_requestId, _timestamp));
         voteCount++;
         uint256 _id = voteCount;
         voteRounds[_hash].push(_id);
         if (voteRounds[_hash].length > 1) {
             uint256 _prevId = voteRounds[_hash][voteRounds[_hash].length - 2];
-            require(block.timestamp - voteInfo[_prevId].startDate > 12 hours, "this dispute is still open");
```
Comment (tallywiesenberg) [inline contracts/Governance.sol]: just double checking, this is correct?

File: contracts/Governance.sol (hunk 2 above)
Comment (tallywiesenberg) [reply contracts/Governance.sol]: agreed

**Comment by tallywiesenberg** [comment]
I removed those edits to the Governance contract

**Review by themandalore — APPROVED** [review]

**Review by themandalore — APPROVED** [review]

**Pull request merged**